File: src/Lessons/4-CountingElements/4.4-MissingIntegers.js
function solution(A) {
    //Find missing integers within the array
    let array = [... A.sort((a, b) => a - b)];

    let missingInteger = 1;
    for (let index = 0; index < array.length; index++) {
        const integer = array[index];

        if(integer > 0){
            if(integer != missingInteger){
                if(array[index-1] == missingInteger)
                    missingInteger++;
                else
                    break;
            }
        }     
    }
    if(missingInteger == array[array.length-1]) missingInteger++;
    return missingInteger;
}

const testArray = [-2,-1];
console.log(solution(testArray))
//expect 7

File: src/Lessons/4-CountingElements/4.4-MissingIntegers.go
package main

import (
	"fmt"
	"sort"
)

func main() {

	fmt.Println(solution([]int{-2, -5, 0}))
}

func solution(someArray []int) int {
	//Return the smallest positive integer that does not occur in A
	missingInteger := 1
	//Sort our array
	sort.Ints(someArray)
	for index := 0; index < len(someArray); index++ {
		num := someArray[index]
		if num > 0 {
			if num != missingInteger {
				//Check if previous number matches the missing
				if index < 1 {
					break
				}
				prevNumber := someArray[index-1]
				if prevNumber == missingInteger {
					missingInteger++
				} else {
					//if not we've found the missing number break
					break
				}
			}
		}
	}
	//incase no missing integer was found
	if missingInteger == someArray[len(someArray)-1] {
		missingInteger++
	}

	return missingInteger
}
